perun-registrar-lib/src/main/java/cz/metacentrum/perun/registrar/impl/ConsolidatorManagerImpl.java

```diff
@@ -460,6 +458,184 @@ public List<UserExtSource> consolidateIdentityUsingToken(PerunSession sess, Stri
 
 	}
 
+	@Override
+	public Void consolidate(PerunSession sess, String accessToken) throws PerunException {
+		OidcUserInfo userInfo2 = getOidcSubIssuer(accessToken, BeansUtils.getCoreConfig().getOidcIssuersExtsourceNames().keySet().iterator().next());
+
+		User user1 = sess.getPerunPrincipal().getUser();
+
+		ExtSource extSource = perun.getExtSourcesManagerBl().getExtSourceByName(sess, userInfo2.issuer);
+		UserExtSource ues = new UserExtSource();
+		ues.setLogin(userInfo2.sub);
+		ues.setExtSource(extSource);
+		User user2;
+		try {
+			user2 = perun.getUsersManagerBl().getUserByUserExtSource(sess, ues);
+		} catch (UserNotExistsException ex) {
+			user2 = null;
+		}
+
+		log.info("User who called consolidation is: {}, Previously logged user is: {}", user1, user2);
+
+		if(userInfo2.issuer.equals(sess.getPerunPrincipal().getExtSourceName()) &&
+			userInfo2.sub.equals(sess.getPerunPrincipal().getActor())) {
+			IdentityIsSameException ex = new IdentityIsSameException("You tried to join same identity with itself. Please try again but select different identity.");
+			log.warn("User tried to join identity with itself.");
+			throw ex;
+		}
+
+		if(user1 == null && user2 == null) {
+			IdentityUnknownException ex = new IdentityUnknownException("Neither original or current identity is know to Perun. Please use at least one identity known to Perun.");
+			ex.setLogin(ues.getLogin());
+			ex.setSource2(ues.getExtSource().getName());
+			ex.setSourceType2(ues.getExtSource().getType());
+			ex.setLogin2(sess.getPerunPrincipal().getActor());
+			ex.setSource2(sess.getPerunPrincipal().getExtSourceName());
+			ex.setSourceType2(sess.getPerunPrincipal().getExtSourceType());
+			log.warn("None of identities is known to Perun. Current identity: {}, previous identity: {}",
+				StringUtils.join(Arrays.asList(ex.getLogin(), ex.getSource(), ex.getSourceType()), " | "),
+				StringUtils.join(Arrays.asList(ex.getLogin2(), ex.getSource2(), ex.getSourceType2()), " | "));
+			throw ex;
+		} else if(user1 != null && user2 != null) {
```

Comment:
check if the two identites are registered - if both are registered the account linking wont end succesfully